Sources/Common/DataModel/DataSet/index.js

```diff
@@ -57,6 +28,87 @@ function vtkDataSet(publicAPI, model) {
     }
   });
 
+  //------------------------------------------------------------------------------
+  // Compute the data bounding box from data points.
+  publicAPI.computeBounds = () => {
+    if (
+      (model.modifiedTime &&
+        model.computeTime &&
+        model.modifiedTime > model.computeTime) ||
+      !model.computeTime
+    ) {
+      const points = publicAPI.getPoints();
+      if (points?.getNumberOfPoints()) {
+        // Compute bounds from points
+        const bounds = vtkBoundingBox.INIT_BOUNDS.slice();
+        const numberOfPoints = points.getNumberOfPoints();
+        for (let i = 0; i < numberOfPoints; i++) {
+          const pt = points.getPoint(i);
+          vtkBoundingBox.addPoint(bounds, pt[0], pt[1], pt[2]);
+        }
+        model.bounds = bounds;
+      } else {
+        model.bounds = vtkMath.createUninitializedBounds();
+      }
+      // Update computeTime
+      model.computeTime = Date.now();
+    }
+  };
+
+  /**
+   * Returns the bounds as [xmin, xmax, ymin, ymax, zmin, zmax]
+   */
+  publicAPI.getBounds = () => {
+    if (model.bounds && model.bounds.length === 6) {
+      return model.bounds;
+    }
+    // fallback: try to get bounds from pointData if available
+    if (publicAPI.getPoints && publicAPI.getPoints()) {
+      return publicAPI.getPoints().getBounds();
+    }
+    return vtkMath.createUninitializedBounds();
+  };
```

Comment:
Actually, scratch all that, there is no "getPoints()" in a DataSet, so it's pointless to compute bounds in here.